specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+# Background
+Our end developers have pointed out gaps in the existing CoreWebView2.ExecuteScript method, and it is necessary
+to provide a new method to let our end developers get more information in a more convenient manner.
+The new ExecuteScriptWithResult method will provide exception information if the executed script
+failed, and provides a new method to try to get the script execution result as a string rather than as JSON
+in order to make it more convenient to interact with string results.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` to provide an `ExecuteScriptWithResult` 
```

Comment:
ExecuteScriptWithResult feels slightly confusing/redundant with ExecuteScriptAsync since the existing one already returns a Task<string> that is called the "result".  Are we proposing an alternate function name because an overload can't be differentiated only by return type, or are there also other reasons?

Reply:
Yeah, the WithResult hear means that this API user will get all the results of executing JavaScript, including exceptions, And I would think the ExecuteScript can be considered to be marked as deprecated after this API is published.

Reply:
Thanks.  This aligns with my expectations, just a little weird in this case since Blah() already returned something called the "result". 🙂